crates/provider/src/utils.rs

```
@@ -39,6 +39,17 @@ pub fn eip1559_default_estimator(base_fee_per_gas: U256, rewards: &[Vec<U256>])
     (potential_max_fee, max_priority_fee_per_gas)
 }
 
+/// Identifies the intended transport type from the given string.
+pub fn parse_str_to_tranport_type(s: &str) -> &str {
```

this should return an error instead of invalid output